src/components/Modal/SbModal.vue

```diff
@@ -0,0 +1,105 @@
+<template>
+  <SbPortal
+    ref="portalRef"
+    slim
+    append
+    unmount-on-destroy
+    target-slim
+    :target="modalTarget"
+  >
+    <SbBlokUi v-if="open" @click="wrapClose">
+      <div
+        ref="modal"
+        class="sb-modal"
+        :class="{ 'sb-modal__full-width': fullWidth }"
+        role="dialog"
+        v-bind="{ ...$attrs }"
+      >
+        <button
+          class="sb-modal__close-button"
+          aria-label="Close Modal"
+          @click="handleCloseModal"
+        >
+          <SbIcon name="close" size="normal" color="primary-dark" />
+        </button>
+        <slot />
+      </div>
+    </SbBlokUi>
+  </SbPortal>
+</template>
+
+<script>
+import SbIcon from '../Icon'
+import SbBlokUi from '../BlockUI'
+import ClickOutside from '../../directives/click-outside'
+import SbPortal from '../Portal'
+import { randomString } from '../../utils'
+
+export default {
+  name: 'SbModal',
+
+  components: {
+    SbIcon,
+    SbPortal,
+    SbBlokUi,
+  },
+
+  directives: {
+    ClickOutside,
+  },
+
+  props: {
+    isOpenModal: Boolean,
+    fullWidth: Boolean,
+    modalTarget: {
+      type: String,
+      default: () => `#sb-portal-target-${randomString(4)}`,
+    },
+  },
+
+  data() {
+    return {
+      open: false || this.isOpenModal,
+    }
+  },
+
+  watch: {
+    isOpenModal: function () {
```

Comment:
isOpenModal() to pattern?

Reply:
The **watcher** `isOpenModal` is to ensure that if the modal is called more than once on the same component it behaves the way you want, opening and closing when called.

Reply:
Sorry, I think you don't understand. How about use `isOpenModal()` instead of `isOpenModal: function ()` to maintain the pattern?

Reply:
Ahhh yes, you are correct thanks 😄